hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/multitenant/AuthorizerLockImpl.java

```diff
@@ -132,11 +132,12 @@ public long tryWriteLockThrowOnTimeout() throws IOException {
   @Override
   public void tryWriteLockInOMRequest() throws IOException {
 
+    long stamp = tryWriteLockThrowOnTimeout();
+
     // Sanity check. Must not have held a write lock in a tenant OMRequest.
     Preconditions.checkArgument(omRequestWriteLockStamp == 0L);
```

Comment:
It would make sense to check inside a try and release the lock on failure. Also, `validateAndUpdateCache` has a pre-condition check which might also land up not unlocking on exception, might make sense to clean up that code as well.

Reply:
In the case of a bug causing the `Precondition` to fail, we are still holding with the lock when an `IllegalArgumentException` is thrown due to the check having failed. It is a minor nit and unlikely to occur but from a completeness stand point, we should release the lock.

Reply:
> In the case of a bug causing the Precondition to fail

I agree that if the `Precondition` in `tryWriteLockInOMRequest` failed, the lock won't be released. Actually the OM will be in serious trouble if the `Precondition` check doesn't pass (e.g. when some one / some other thread is using the lock incorrectly). Honestly I think should just intentionally crash the OM at this point because OM is likely running in an undefined / unrecoverable state at this point, and the sooner it crashes the easier to debug the problem (and prevent the unknown error from propagating).

Reply:
@smengcl this is a bug around tenant administration, not sure if crashing OM would be the right way. That said, I agree that if precondition were to fail, it would imply a serious coding issue here... At a minimum we need to clean up the precondition check in `validateAndUpdateCache` and insure that the lock is unlocked.